sdk/cosmos/azure-cosmos/samples/document_management.py

```diff
@@ -310,6 +444,21 @@ def run_sample():
         delete_all_items_by_partition_key(db, "CompanyA")
         query_items_with_continuation_token_size_limit(container, 'SalesOrder1')
 
+        # setup MultiHash samples
+        container_multi_hash = db.create_container_if_not_exists(id=CONTAINER_MH_ID,
```

Comment:
I terms of new public API surface area, is it basically:
- the `PartitionKey` object now accepts a list or paths and a `kind` parameter
- the `partition_key` parameter noew accepts a list - or was this always the case?

Reply:
For the former, PartitionKey has always been able to accept a kind parameter it just always defaults to Hash since that was the only value that would not result in an error. For Paths it can be any value, originally the value given put just be made the only value in the Paths property(a list) of the PartitionKey, now if the kind is set to MultiHash, Paths will be a list of up to three values that represent the levels of the MultiHash PartitionKeys.

for the latter yes partition_key parameter now accepts a list if the container was set up with subpartitioning.

Reply:
In that case every typehint/docstring for `partition_key` will need to be updated to list support for `List[str]`. (I would wait until the pylint PR is merged before doing that though - or you will end up with merge headaches.)
Also, in the case of the `PartitionKey` object constructor, we can create a better experience for this using overloads:
```python
@overload
def __init__(self, path: List[str], *, kind: Literal["MultiHash"], version: int = 2) -> None:
    ...
@overload
def __init__(self, path: str, *, kind: Literal["Hash"] = "Hash", version: int = 2) -> None:
    ...

def __init__(self, *args, **kwargs):
        self.path = args[0] if args else kwargs['path']
        self.kind = args[1] if len(args) > 1 else kwargs.get('kind', 'Hash')
        self.version = args[2] if len(args) > 2 else kwargs.get('version', 2)
```

Reply:
The partition key path should support a list of non-string values such as ints and floats. As well as mix value types. I have made the other recommended changes and made sure it's compatible with all supported python versions.